payu/models/staged_cable.py

```diff
@@ -0,0 +1,205 @@
+        cable_stages = self._prepare_configuration()
+
+        # Build the configuration log
+        self.configuration_log['queued_stages'] = cable_stages
+        self.configuration_log['current_stage'] = ''
+        self.configuration_log['completed_stages'] = []
+
+        with open('configuration_log.yaml', 'w') as conf_log_file:
+            yaml.dump(self.configuration_log, conf_log_file)
+
+    def _read_configuration_log(self):
+        """Read the existing configuration log."""
+        with open('configuration_log.yaml') as conf_log_file:
+            self.configuration_log = yaml.safe_load(conf_log_file)
+
+    def _prepare_configuration(self):
+        """Prepare the stages in the CABLE configuration."""
+
+        # Since Python3.7, dictionary order is guaranteed so we can read
+        # the entries in order without needing to supply an index
+        # We just want to populate cable_stages with the list of stages
+        # to run
+        cable_stages = []
+        for stage_name, stage_opts in self.stage_config.items():
+            # Check if stage is a multi-step or single step
+            if stage_name.startswith('multistep'):
+                # The multi-step stage can run each internal stage
+                # a different number of times. For example, a two
+                # step stage may ask for the first step (S1) 5 times,
+                # but the second step (S2) only 3 times. The stage
+                # looks like [S1, S2, S1, S2, S1, S2, S1, S1].
+                # So what we need to do is first record the number
+                # of times each step is run
+                step_names = []
+                step_counts = []
+                for step_name, step_opts in stage_opts.items():
+                    step_names.append(step_name)
+                    step_counts.append(step_opts['count'])
+
+                # Now iterate to the maximum number of steps
+                for step in range(max(step_counts)):
+                    for step_id, _ in enumerate(step_names):
+                        if step_counts[step_id] > step:
+                            cable_stages.append(step_names[step_id])
+            # Finish handling of multistep stage
+
+            else:
+                # A single step stage, in general we only want to run this
+                # once, but check for the count anyway
+                for _ in range(stage_opts['count']):
+                    cable_stages.append(stage_name)
+
+        # Finish handling of single step stage
+        return cable_stages
+
+    def setup(self):
+        super(StagedCable, self).setup()
+
+        # Directories required by CABLE for outputs
+        os.makedirs(os.path.join(self.work_output_path, 'logs'),
+                    exist_ok=True)
+        os.makedirs(os.path.join(self.work_output_path, 'restart'),
+                    exist_ok=True)
+        os.makedirs(os.path.join(self.work_output_path, 'outputs'),
+                    exist_ok=True)
```

Comment:
Could reduce code duplication
```suggestion
        for dir in ['logs', 'restart', 'outputs']:
           os.makedirs(os.path.join(self.work_output_path, dir),
                    exist_ok=True)
```